File: benchs/object/object.ml

class point x y = object
  val x = x
  val y = y
  method sym = (new point (-x) (-y))
end

let _ =
  let start = millis () in
  for _ = 1 to 10 do
    for _ = 1 to 10 do
      let o = new point (Random.int 100) (Random.int 100) in
      for _ = 1 to 10 do
        ignore(o#sym)
      done
    done
  done;
  let stop = millis () in ()
  (* Serial.write_string (string_of_int (stop-start)) *)

File: benchs/object/object.py
import time
import random
class Point:
    def __init__(self, x,y):
        self.x = x
        self.y = y
    
    def sym(self):
        return (Point(-self.x,-self.y))

def main():
    start = time.ticks_ms()
    for i in range(0,10):
        for j in range(0,10):
            o = Point(random.randint(0,100),random.randint(0,100))
            for k in range (0,10):
                o.sym()
    stop=time.ticks_ms()
    print(time.ticks_diff(stop,start))

if __name__ == "__main__":
    main()
